src/task.cpp

```diff
@@ -725,7 +725,7 @@ TError TTask::ChildSetHostname() {
         string host = Env->Hostname + "\n";
         TError error = f.WriteStringNoAppend(host);
         if (error)
-            return TError(EError::Unknown, error, "write(/etc/hostname)");
```

Comment:
And this is another interesting line.
Operator bool returns only true or false and the real errno value will be lost.